README
NAME
    Task::Perl::Critic::IncludingOptionalDependencies - Install everything
    Perl::Critic plus its optional dependencies.

VERSION
    This document describes
    Task::Perl::Critic::IncludingOptionalDependencies version 1.4.0.

SYNOPSIS
    This module does nothing but act as a placeholder. See Task.

DESCRIPTION
    This module does nothing but act as a placeholder. See Task.

    Inspired by a question at the Perl::Critic BoF at OSCON 2007.

    WARNING: Installing this distribution will install Policies that
    directly conflict with each other. If you do not use a .perlcriticrc
    file, and your severity is set high enough, there is no way for your
    code to not have violations. A specific example:
    Perl::Critic::Policy::Compatibility::ProhibitThreeArgumentOpen and
    Perl::Critic::Policy::InputOutput::ProhibitTwoArgOpen directly
    contradict each other.

INTERFACE
    None.

DIAGNOSTICS
    None.

CONFIGURATION AND ENVIRONMENT
    Task::Perl::Critic::IncludingOptionalDependencies requires no
    configuration files or environment variables.

DEPENDENCIES
    Task::Perl::Critic

    File::HomeDir

    File::Which

    IPC::Open2

    Perl::MinimumVersion

    Perl::Tidy

    Pod::Spell

    Readonly::XS

    Regexp::Parser

    Term::ANSIColor

    Text::ParseWords

INCOMPATIBILITIES
    None reported.

BUGS AND LIMITATIONS
    No bugs have been reported.

    Please report any bugs or feature requests to
    "bug-task-perl-critic-includingoptionaldependencies@rt.cpan.org", or
    through the web interface at <http://rt.cpan.org>.

AUTHOR
    Elliot Shank "<perl@galumph.com>"

LICENSE AND COPYRIGHT
    Copyright (C)2007, Elliot Shank "<perl@galumph.com>". All rights
    reserved.

    This module is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself. See perlartistic.

DISCLAIMER OF WARRANTY
    BECAUSE THIS SOFTWARE IS LICENSED FREE OF CHARGE, THERE IS NO WARRANTY
    FOR THE SOFTWARE, TO THE EXTENT PERMITTED BY APPLICABLE LAW. EXCEPT WHEN
    OTHERWISE STATED IN WRITING THE COPYRIGHT HOLDERS AND/OR OTHER PARTIES
    PROVIDE THE SOFTWARE "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER
    EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
    ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE OF THE SOFTWARE IS WITH
    YOU. SHOULD THE SOFTWARE PROVE DEFECTIVE, YOU ASSUME THE COST OF ALL
    NECESSARY SERVICING, REPAIR, OR CORRECTION.

    IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
    WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR
    REDISTRIBUTE THE SOFTWARE AS PERMITTED BY THE ABOVE LICENSE, BE LIABLE
    TO YOU FOR DAMAGES, INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL, OR
    CONSEQUENTIAL DAMAGES ARISING OUT OF THE USE OR INABILITY TO USE THE
    SOFTWARE (INCLUDING BUT NOT LIMITED TO LOSS OF DATA OR DATA BEING
    RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD PARTIES OR A
    FAILURE OF THE SOFTWARE TO OPERATE WITH ANY OTHER SOFTWARE), EVEN IF
    SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH
    DAMAGES.



Changes
[1.004] Released on 2008-07-06

    "use 5.006001" in Makefile.PL and Build.PL.

    Add Readonly::XS.

[1.003] Released on 2008-06-08.

    Documentation updates.

    Replace use of Module::Install with Module::Build.

[1.002] Released on 2008-03-15.

    Add new optional dependencies in Perl::Critic 1.082.

[1.001] Released on 2007-09-23.

    Fix the distribution name.  Whoops.

    Update Task::Perl::Critic dependency version.

[1.000] Released on 2007-07-30.

    Initial version.

#      $URL: http://perlcritic.tigris.org/svn/perlcritic/trunk/Task-Perl-Critic-IncludingOptionalDependencies/Changes $
#     $Date: 2008-07-06 07:17:47 -0500 (Sun, 06 Jul 2008) $
#   $Author: clonezone $
# $Revision: 2502 $


Makefile.PL
# Note: this file was auto-generated by Module::Build::Compat version 0.2808_01
use ExtUtils::MakeMaker;
WriteMakefile
(
          'PL_FILES' => {},
          'INSTALLDIRS' => 'site',
          'NAME' => 'Task::Perl::Critic::IncludingOptionalDependencies',
          'EXE_FILES' => [],
          'VERSION_FROM' => 'lib/Task/Perl/Critic/IncludingOptionalDependencies.pm',
          'PREREQ_PM' => {
                           'Term::ANSIColor' => 0,
                           'File::HomeDir' => 0,
                           'File::Which' => 0,
                           'Perl::Tidy' => 0,
                           'Regexp::Parser' => '0.2',
                           'Pod::Spell' => 1,
                           'Perl::MinimumVersion' => 0,
                           'Test::More' => 0,
                           'Text::ParseWords' => 3,
                           'IPC::Open2' => 1,
                           'Task::Perl::Critic' => '1.006',
                           'Readonly::XS' => 0
                         }
        )
;


lib/Task/Perl/Critic/IncludingOptionalDependencies.pm
package Task::Perl::Critic::IncludingOptionalDependencies;

use 5.006001;
use strict;
use warnings;

our $VERSION = '1.004';


1; # Magic true value required at end of module

__END__

=for stopwords BoF OSCON

=head1 NAME

Task::Perl::Critic::IncludingOptionalDependencies - Install everything Perl::Critic plus its optional dependencies.


=head1 VERSION

This document describes
Task::Perl::Critic::IncludingOptionalDependencies version 1.4.0.


=head1 SYNOPSIS

This module does nothing but act as a placeholder.  See L<Task>.


=head1 DESCRIPTION

This module does nothing but act as a placeholder.  See L<Task>.

Inspired by a question at the Perl::Critic BoF at OSCON 2007.

B<WARNING>: Installing this distribution will install Policies that
directly conflict with each other.  If you do not use a
F<.perlcriticrc> file, and your severity is set high enough, there is
no way for your code to not have violations.  A specific example:
L<Perl::Critic::Policy::Compatibility::ProhibitThreeArgumentOpen> and
L<Perl::Critic::Policy::InputOutput::ProhibitTwoArgOpen> directly
contradict each other.


=head1 INTERFACE

None.


=head1 DIAGNOSTICS

None.


=head1 CONFIGURATION AND ENVIRONMENT

Task::Perl::Critic::IncludingOptionalDependencies requires no
configuration files or environment variables.


=head1 DEPENDENCIES

L<Task::Perl::Critic>

L<File::HomeDir>

L<File::Which>

L<IPC::Open2>

L<Perl::MinimumVersion>

L<Perl::Tidy>

L<Pod::Spell>

L<Readonly::XS>

L<Regexp::Parser>

L<Term::ANSIColor>

L<Text::ParseWords>


=head1 INCOMPATIBILITIES

None reported.


=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests to
C<bug-task-perl-critic-includingoptionaldependencies@rt.cpan.org>, or
through the web interface at L<http://rt.cpan.org>.


=head1 AUTHOR

Elliot Shank  C<< <perl@galumph.com> >>


=head1 LICENSE AND COPYRIGHT

Copyright (C)2007, Elliot Shank C<< <perl@galumph.com> >>. All rights
reserved.

This module is free software; you can redistribute it and/or modify it
under the same terms as Perl itself. See L<perlartistic>.


=head1 DISCLAIMER OF WARRANTY

BECAUSE THIS SOFTWARE IS LICENSED FREE OF CHARGE, THERE IS NO WARRANTY
FOR THE SOFTWARE, TO THE EXTENT PERMITTED BY APPLICABLE LAW. EXCEPT
WHEN OTHERWISE STATED IN WRITING THE COPYRIGHT HOLDERS AND/OR OTHER
PARTIES PROVIDE THE SOFTWARE "AS IS" WITHOUT WARRANTY OF ANY KIND,
EITHER EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
PURPOSE. THE ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE OF THE
SOFTWARE IS WITH YOU. SHOULD THE SOFTWARE PROVE DEFECTIVE, YOU ASSUME
THE COST OF ALL NECESSARY SERVICING, REPAIR, OR CORRECTION.

IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR
REDISTRIBUTE THE SOFTWARE AS PERMITTED BY THE ABOVE LICENSE, BE LIABLE
TO YOU FOR DAMAGES, INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL, OR
CONSEQUENTIAL DAMAGES ARISING OUT OF THE USE OR INABILITY TO USE THE
SOFTWARE (INCLUDING BUT NOT LIMITED TO LOSS OF DATA OR DATA BEING
RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD PARTIES OR A
FAILURE OF THE SOFTWARE TO OPERATE WITH ANY OTHER SOFTWARE), EVEN IF
SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH
DAMAGES.

=cut

# Local Variables:
#   mode: cperl
#   cperl-indent-level: 4
#   fill-column: 70
#   indent-tabs-mode: nil
#   c-indentation-style: bsd
# End:
# ex: set ts=8 sts=4 sw=4 tw=70 ft=perl expandtab :


Build.PL
#!perl

#      $URL: http://perlcritic.tigris.org/svn/perlcritic/trunk/Task-Perl-Critic-IncludingOptionalDependencies/Build.PL $
#     $Date: 2008-07-06 07:11:00 -0500 (Sun, 06 Jul 2008) $
#   $Author: clonezone $
# $Revision: 2500 $

use 5.006001;

use strict;
use warnings;

use English qw< -no_match_vars >;

use Module::Build;


my $class = Module::Build->subclass( code => <<'END_SUBCLASS' );

    sub ACTION_test {
        my ($self) = @_;

        $self->depends_on('manifest');

        return $self->SUPER::ACTION_test();
    }

    sub ACTION_authortest {
        my ($self) = @_;

        $self->depends_on('build');
        $self->depends_on('manifest');
        $self->depends_on('distmeta');

        $self->test_files( qw< t xt/author > );
        $self->recursive_test_files(1);

        $self->depends_on('test');

        return;
    } # end ACTION_authortest()

    sub ACTION_distdir {
        my ($self) = @_;

        $self->depends_on('authortest');

        return $self->SUPER::ACTION_distdir();
    } # end ACTION_distdir

END_SUBCLASS

my $builder = $class->new(
    module_name         => 'Task::Perl::Critic::IncludingOptionalDependencies',
    dist_author         => 'Elliot Shank <perl@galumph.com>',
    license             => 'perl',
    dynamic_config      => 0,
    create_readme       => 1,
    create_packlist     => 1,
    sign                => 0,

    requires            => {
        'Task::Perl::Critic'   => 1.006,
        'File::HomeDir'        => 0,
        'File::Which'          => 0,
        'IPC::Open2'           => 1,
        'Perl::MinimumVersion' => 0,
        'Perl::Tidy'           => 0,
        'Pod::Spell'           => 1,
        'Readonly::XS'         => 0,
        'Regexp::Parser'       => 0.20,
        'Term::ANSIColor'      => 0,
        'Text::ParseWords'     => 3,
    },

    build_requires      => {
        'Test::More'    => 0,
    },

    recommends          => {
    },

    create_packlist     => 1,

    sign                => 0,

    create_makefile_pl  => 'traditional',

    meta_merge          => {
        resources   => {
            homepage    => 'http://perlcritic.com',
            license     => 'http://dev.perl.org/licenses/',
            Repository  => 'http://perlcritic.tigris.org/svn/perlcritic/trunk/Task-Perl-Critic/',
            MailingList => 'http://perlcritic.tigris.org/servlets/SummarizeList?listName=users',
        }
    },

    add_to_cleanup      => [
        qw<
            MANIFEST
            MANIFEST.bak
            META.yml
            Makefile
            Makefile.old
            Makefile.PL
            pod2htm?.tmp
            pm_to_blib
            README
            Task-Perl-Critic-*
        >,
    ],
);


$builder->create_build_script();

##############################################################################
# Local Variables:
#   mode: cperl
#   cperl-indent-level: 4
#   fill-column: 78
#   indent-tabs-mode: nil
#   c-indentation-style: bsd
# End:
# setup vim: set filetype=perl tabstop=4 softtabstop=4 expandtab :
# setup vim: set shiftwidth=4 shiftround textwidth=78 nowrap autoindent :
# setup vim: set foldmethod=indent foldlevel=0 :
